From f6d47db48ee36a03260958756fa9fe0bb3ee7604 Mon Sep 17 00:00:00 2001
From: tstegema <tstegema@90b65887-3827-0410-9a23-83215b262276>
Date: Mon, 15 Sep 2014 16:15:12 +0000
Subject: [PATCH 0577/1077] MANTIS 0033581 extension: Desko-Penta: support RFID

massai_systools.lib
V1.0.0.21

SCardDisconnect results in SCARD_STATE_MUTE and SCARD_STATE_PRESENT


git-svn-id: svn://localhost/SelfServiceCommon/trunk@726 90b65887-3827-0410-9a23-83215b262276
---
 .../Massai/cpp/Systools/src/mSysSmartCardReaderW32.cpp         | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/SelfServiceCommon/Massai/cpp/Systools/src/mSysSmartCardReaderW32.cpp b/SelfServiceCommon/Massai/cpp/Systools/src/mSysSmartCardReaderW32.cpp
index 9b673d77..18569479 100644
--- a/SelfServiceCommon/Massai/cpp/Systools/src/mSysSmartCardReaderW32.cpp
+++ b/SelfServiceCommon/Massai/cpp/Systools/src/mSysSmartCardReaderW32.cpp
@@ -318,11 +318,12 @@ namespace
   {
     trc.writeTrace(BasicRuntime_HERE, ">SCardDisconnect");
     protocol = SCARD_PROTOCOL_UNDEFINED;
-    LONG errorCode = SCardDisconnect(card, SCARD_EJECT_CARD);
+    LONG errorCode = SCardDisconnect(card, SCARD_LEAVE_CARD);
     card = 0;
     protocol = SCARD_PROTOCOL_UNDEFINED;
     trc.writeTrace(BasicRuntime_HERE, "<SCardDisconnect:(0x%x)(%s)",
       errorCode, getErrorCodeText(errorCode).c_str());
+    cardStatus |= SCARD_STATE_MUTE | SCARD_STATE_PRESENT;
     return errorCode;
   }
 
-- 
2.41.0.windows.1

